crates/bevy_pbr/src/deferred/mod.rs

```diff
@@ -0,0 +1,451 @@
+use crate::{MeshPipeline, MeshViewBindGroup, ScreenSpaceAmbientOcclusionSettings};
+use bevy_app::prelude::*;
+use bevy_asset::{load_internal_asset, Handle};
+use bevy_core_pipeline::{
+    clear_color::ClearColorConfig,
+    core_3d,
+    deferred::{
+        copy_lighting_id::DeferredLightingIdDepthTexture, DEFERRED_LIGHTING_PASS_ID_DEPTH_FORMAT,
+    },
+    prelude::{Camera3d, ClearColor},
+    prepass::DeferredPrepass,
+    tonemapping::{DebandDither, Tonemapping},
+};
+use bevy_ecs::{prelude::*, query::QueryItem};
+use bevy_render::{
+    extract_component::{
+        ComponentUniforms, ExtractComponent, ExtractComponentPlugin, UniformComponentPlugin,
+    },
+    render_asset::RenderAssets,
+    render_graph::{NodeRunError, RenderGraphContext, ViewNode, ViewNodeRunner},
+    render_resource::{self, Operations, PipelineCache, RenderPassDescriptor},
+    renderer::{RenderContext, RenderDevice},
+    texture::Image,
+    view::{ViewTarget, ViewUniformOffset},
+    Render, RenderSet,
+};
+
+use bevy_render::{
+    render_graph::RenderGraphApp, render_resource::*, texture::BevyDefault, view::ExtractedView,
+    RenderApp,
+};
+
+use crate::{
+    EnvironmentMapLight, MeshPipelineKey, ViewFogUniformOffset, ViewLightsUniformOffset,
+    MAX_CASCADES_PER_LIGHT, MAX_DIRECTIONAL_LIGHTS,
+};
+
+pub struct PbrDeferredLightingPlugin;
+
+pub const DEFERRED_LIGHTING_SHADER_HANDLE: Handle<Shader> =
+    Handle::weak_from_u128(2708011359337029741);
+
+pub const DEFAULT_PBR_DEFERRED_LIGHTING_DEPTH_ID: u8 = 1;
+
+/// Component with a `depth_id` for specifying which corresponding materials should be rendered by this specific PBR deferred lighting pass.
+/// Will be automatically added to entities with the [`DeferredPrepass`] component that don't already have a [`PbrDeferredLightingDepthId`].
+#[derive(Component, Clone, Copy, ExtractComponent, ShaderType)]
+pub struct PbrDeferredLightingDepthId {
+    depth_id: u32,
+
+    #[cfg(all(feature = "webgl", target_arch = "wasm32"))]
+    _webgl2_padding: bevy_math::Vec3,
```

Comment:
```suggestion
    #[cfg(all(feature = "webgl", target_arch = "wasm32"))]
    _webgl2_padding_0: f32,
    #[cfg(all(feature = "webgl", target_arch = "wasm32"))]
    _webgl2_padding_1: f32,
    #[cfg(all(feature = "webgl", target_arch = "wasm32"))]
    _webgl2_padding_2: f32,
```
or some better way of applying the cfg to all three.